src/main/java/com/artipie/file/FileProxy.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.file;
+
+import com.artipie.RepoConfig;
+import com.artipie.files.FileProxySlice;
+import com.artipie.http.Response;
+import com.artipie.http.Slice;
+import com.artipie.http.client.ClientSlices;
+import com.artipie.repo.ProxyConfig;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Map;
+import org.reactivestreams.Publisher;
+
+/**
+ * File proxy slice created from config.
+ *
+ * @since 0.12
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
```

Comment:
@olegmoz is this suppression necessary here? I'd suggest not to add it if not, as generally this [check](https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.html) is connected to the complexity of the class and when we have to add this suppression it's the first signal to required decomposition.

Reply:
@olenagerasimova that suppression was here by mistake, I totally agree that this warning is meaningful in production code